ways_of_working/processes/version_control.md

```diff
@@ -0,0 +1,96 @@
+---
+notion_page: https://www.notion.so/wpmedia/Version-control-94bd0c31e0514e31a74c00241f103690?pvs=4
+title: Version Control
+---
+
+# Version Control
+
+## Git
+
+At WP Media, we collaborate on code using [Git](https://git-scm.com/). This is a powerful and standard version control system that allows parallel work and offers codebase history tracking.
+Our main platform for this is [GitHub](https://github.com/wp-media), while some infrastructure-related repositories are handled on our internal GitLab.
+
+Git is a powerful system that supports many workflows and ways of using it. To ease collaboration and clarify expectations, we prefer to work with a standardized flow at WP Media. The following chapters describe the main guidelines about our standard Git management.
+
+### Branch management
+
+#### Long-lived branches
+
+Our standard repository configuration is to have two main branches: `trunk` and `develop`:
+- `trunk` updates should correspond to releases. The current state of trunk should therefore be the codebase currently in production (for services) or the latest released version (for plugins). Pushes to trunk are related to a release process. For instance, a CI/CD process can be automated based on this action ; or releases can be tagged based on this branch. 
+- `develop` should be the main branch from which development branches will start from, and be merged into.
+
+`develop` should always be strictly ahead of `trunk`. If it is not the case, `trunk` must be merged back into `develop`.
+
+Additional long-lived branches might exist for specific needs, for instance for `transifex` ; but this must remain an exception.
+
+#### Development branches
+
+To contribute to a repository, developers should always create a new branch from develop and commit their changes there. Once the implementation is ready to be reviewed and validated, they should open a Pull Request to develop. 
```

Comment:
I didn't see your PR @MathieuLamiot 

Can we add something about rebasing develop before starting a new branch?

Reply:
On most repositories, I don't think we rebase develop 🤔 Worst case scenario, trunk is ahead of develop (which should not happen anyway) and we can merge trunk into develop. What do you have in mind exactly?

Reply:
If you don't update your develop on local, when merging into trunk (release), it takes previous merge requests.
See the 3 consecutive releases on wp-rocket.me:
- https://github.com/wp-media/wp-rocket.me/pull/4095
- https://github.com/wp-media/wp-rocket.me/pull/4099
- https://github.com/wp-media/wp-rocket.me/pull/4102

The PR #4088 (Coupon Links Woo HPOS compat) was listed 3 times.
It then creates erroneous changelogs.
What do you think about it?

Reply:
ah yeah... squash commits right? I hate squash commits so I tend to forget they exist 😬 rebase is indeed preferred after a squash commit from a long-lived branch (which is not the case with merges).
But it's good practice to squash from develop to trunk ; it's done on wp-rocket plugin too. But we don't do that (or not consistently at least) on other repositories.

Standardization of squash commits for develop and trunk would make sense. But I feel like this requires proper discussion with the whole team first as it can be weird to see/use if you're not use to it. Maybe we can discuss this at the retreat and add this to the engineering book then?
I'm thinking we'd need a GH action to automatically rebase develop on trunk after a squash commit to trunk too 🤔

Reply:
> I'm thinking we'd need a GH action to automatically rebase develop on trunk after a squash commit to trunk too 

I am up for this too.

Let's take the opportunity to discuss common process at the retreat for sure!